solarforecastarbiter/reference_forecasts/persistence.py

```diff
@@ -0,0 +1,360 @@
+# models_persistence.py?
+"""
+Functions for persistence forecasts.
+
+Two kinds of persistence are supported:
+
+  1. Persistence of observed value in :py:func:`persistence`
+  2. Persistence of irradiance or power accounting for solar position in
+     :py:func:`index_persistence`.
+
+The functions take the same parameters and return a forecast. See their
+doc strings for details.
+
+The functions accept a *load_data* keyword argument that allows users to
+change where the functions load the observation data from. This is most
+useful for users that would like to provide their own observation data
+rather than using the solarforecastarbiter database.
+"""
+from functools import partial
+
+import numpy as np
+import pandas as pd
+
+from solarforecastarbiter import datamodel, pvmodel
+from solarforecastarbiter.io.utils import load_data  # does not exist yet
+from solarforecastarbiter.reference_forecasts import forecast
+
+
+def persistence(observation, window, data_start, data_end,
```

Comment:
Interesting to hear you say that because I've been kicking around similar ideas. I've also wondered if two different functions be most clear, e.g. `persistence_scalar` and `persistence_interval`.

Reply:
I like those two names better than `persistence` and `index_persistence`

Reply:
To be clear, I was thinking we'd have 4 functions total:

`persistence_scalar`
`persistence_interval`
`index_persistence_scalar`
`index_persistence_interval`

or maybe 

`persistence_scalar_index`
`persistence_interval_index`

Reply:
`persistence_scalar_index` rather than `index_persistence_scalar` seems to be easier to find.

Also, seems like a good place for a kwarg to indicate index, or no, because `persistence_scalar` and `persistence_scalar_index` will share the same signature and have many lines in common.

Reply:
Ok on the name. 

> and have many lines in common.

Not in the current implementation!

Reply:
Maybe we only need 3 functions:

1. `persistence_scalar`. Data range is used to calculate a single value to persist. Used for intraday forecasts. 
2. `persistence_interval`. Data range and interval length is used to calculate multiple values to persist. Typically used for day ahead forecasts of hourly average power. In principle could be used for intraday forecasts but less likely.
3. `persistence_scalar_index`. Data range is used to calculate a single index value to persist. Used for intraday forecasts when you want to account for solar position. 

I'm less confident about the 4th flavor:

4. `persistence_interval_index`. Data range and interval length is used to calculate multiple index values to persist. For day ahead forecasts, this is an insignificant win over `persistence_interval` because the solar position doesn't change by enough to make a significant difference in forecast accuracy. Maybe the there's a use for intraday forecasts, but I think it's less common and not sure it's worth the effort.